overlord/snapstate/flags.go

```diff
@@ -60,6 +60,10 @@ type Flags struct {
 
 	// IsAutoRefresh is true if the snap is currently auto-refreshed
 	IsAutoRefresh bool `json:"is-auto-refresh,omitempty"`
+
+	// NoReRefresh forces a refresh to skip adding "rerefresh" tasks.
+	// This is important for the Remodel() use-case.
+	NoReRefresh bool `json:"no-rerefresh,omitempty"`
```

Comment:
we should clear this one in ForSnapSetup in flags.go.

Reply:
Indeed, thanks for spotting this. I added the cleanup and also added a test that checks for NoReRefresh correctness and for the cleaning of this flag.